src/core/qgsmapsettings.cpp

```diff
@@ -284,7 +284,8 @@ void QgsMapSettings::setDestinationCrs( const QgsCoordinateReferenceSystem &crs
 {
   mDestCRS = crs;
   mDatumTransformStore.setDestinationCrs( crs );
-
+  setEllipsoid( mDestCRS.ellipsoidAcronym() );
```

Comment:
This should be removed - the ellipsoid should be manually set for the QgsMapSettings by the creator of the QgsMapSettings object.